controlplane/kubeadm/internal/kubeadm_config_map.go

```diff
@@ -196,10 +197,68 @@ func (k *kubeadmConfig) UpdateCoreDNSImageInfo(repository, tag string) error {
 	return nil
 }
 
+// UpdateAPIServer sets the api server configuration to values set in `apiServer` in kubeadm config map.
+func (k *kubeadmConfig) UpdateAPIServer(apiServer kubeadmv1.APIServer) (bool, error) {
+	data, ok := k.ConfigMap.Data[clusterConfigurationKey]
+	if !ok {
+		return false, errors.Errorf("unable to find %q in kubeadm ConfigMap", clusterConfigurationKey)
+	}
+
+	configuration, err := yamlToUnstructured([]byte(data))
+	if err != nil {
+		return false, errors.Wrapf(err, "unable to decode kubeadm ConfigMap's %q to Unstructured object", clusterConfigurationKey)
+	}
+
+	apiServerPath := []string{"apiServer"}
```

Comment:
You can make this a `const` and pass it directly below instead of using an array and `...`